### PR TITLE
JDK-8301497: Replace NULL with nullptr in cpu/s390

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory cpu/s390. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301497](https://bugs.openjdk.org/browse/JDK-8301497): Replace NULL with nullptr in cpu/s390

### Reviewers
 * [Amit Kumar](https://openjdk.org/census#amitkumar) (@offamitkumar - Author)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/12325/head:pull/12325` \
`$ git checkout pull/12325`

Update a local copy of the PR: \
`$ git checkout pull/12325` \
`$ git pull https://git.openjdk.org/jdk.git pull/12325/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12325`

View PR using the GUI difftool: \
`$ git pr show -t 12325`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12325.diff">https://git.openjdk.org/jdk/pull/12325.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/12325#issuecomment-1503046806)

### CODE REVIEW

File: src/hotspot/cpu/s390/frame_s390.cpp [hunk 1]
```diff
@@ -111,7 +111,7 @@ bool frame::safe_for_sender(JavaThread *thread) {
     }
 
     // At this point, there still is a chance that fp_safe is false.
-    // In particular, (fp == NULL) might be true. So let's check and
+    // In particular, (fp == null) might be true. So let's check and
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/frame_s390.cpp:114]: nullptr

File: src/hotspot/cpu/s390/gc/g1/g1BarrierSetAssembler_s390.cpp [hunk 2]
```diff
@@ -404,7 +404,7 @@ void G1BarrierSetAssembler::resolve_jobject(MacroAssembler* masm, Register value
   __ z_braz(Lnot_weak);
   __ verify_oop(value, FILE_AND_LINE);
   DecoratorSet decorators = IN_NATIVE | ON_PHANTOM_OOP_REF;
-  g1_write_barrier_pre(masm, decorators, (const Address*)NULL, value, noreg, tmp1, tmp2, true);
+  g1_write_barrier_pre(masm, decorators, (const Address*)nullptr, value, noreg, tmp1, tmp2, true);
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/gc/g1/g1BarrierSetAssembler_s390.cpp:407]: unnecessary cast?

File: src/hotspot/cpu/s390/interp_masm_s390.cpp [hunk 3]
```diff
@@ -275,7 +275,7 @@ void InterpreterMacroAssembler::check_and_handle_earlyret(Register scratch_reg)
     Register jvmti_thread_state = Z_ARG2;
     Register tmp                = Z_ARG3;
     load_and_test_long(jvmti_thread_state, Address(Z_thread, JavaThread::jvmti_thread_state_offset()));
-    z_bre(L); // if (thread->jvmti_thread_state() == NULL) exit;
+    z_bre(L); // if (thread->jvmti_thread_state() == null) exit;
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/interp_masm_s390.cpp:278]: nullptr

File: src/hotspot/cpu/s390/interp_masm_s390.cpp [hunk 4]
```diff
@@ -981,7 +981,7 @@ void InterpreterMacroAssembler::lock_object(Register monitor, Register object) {
   //   // We stored the monitor address into the object's mark word.
   // } else if (THREAD->is_lock_owned((address)displaced_header))
   //   // Simple recursive case.
-  //   monitor->lock()->set_displaced_header(NULL);
+  //   monitor->lock()->set_displaced_header(null);
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/interp_masm_s390.cpp:984]: nullptr

File: src/hotspot/cpu/s390/interp_masm_s390.cpp [hunk 5]
```diff
@@ -1026,7 +1026,7 @@ void InterpreterMacroAssembler::lock_object(Register monitor, Register object) {
 
   // } else if (THREAD->is_lock_owned((address)displaced_header))
   //   // Simple recursive case.
-  //   monitor->lock()->set_displaced_header(NULL);
+  //   monitor->lock()->set_displaced_header(null);
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/interp_masm_s390.cpp:1029]: nullptr

File: src/hotspot/cpu/s390/interp_masm_s390.cpp [hunk 6]
```diff
@@ -1080,12 +1080,12 @@ void InterpreterMacroAssembler::unlock_object(Register monitor, Register object)
 // else {
   // template code:
   //
-  // if ((displaced_header = monitor->displaced_header()) == NULL) {
-  //   // Recursive unlock. Mark the monitor unlocked by setting the object field to NULL.
-  //   monitor->set_obj(NULL);
+  // if ((displaced_header = monitor->displaced_header()) == null) {
+  //   // Recursive unlock. Mark the monitor unlocked by setting the object field to null.
+  //   monitor->set_obj(null);
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/interp_masm_s390.cpp:1085]: nullptr

File: src/hotspot/cpu/s390/interp_masm_s390.cpp [hunk 7]
```diff
@@ -1080,12 +1080,12 @@ void InterpreterMacroAssembler::unlock_object(Register monitor, Register object)
...
   // } else if (Atomic::cmpxchg(obj->mark_addr(), monitor, displaced_header) == monitor) {
   //   // We swapped the unlocked mark in displaced_header into the object's mark word.
-  //   monitor->set_obj(NULL);
+  //   monitor->set_obj(null);
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/interp_masm_s390.cpp:1088]: nullptr

File: src/hotspot/cpu/s390/interp_masm_s390.cpp [hunk 8]
```diff
@@ -1106,9 +1106,9 @@ void InterpreterMacroAssembler::unlock_object(Register monitor, Register object)
 
   assert_different_registers(monitor, object, displaced_header, current_header);
 
-  // if ((displaced_header = monitor->displaced_header()) == NULL) {
-  //   // Recursive unlock. Mark the monitor unlocked by setting the object field to NULL.
-  //   monitor->set_obj(NULL);
+  // if ((displaced_header = monitor->displaced_header()) == null) {
+  //   // Recursive unlock. Mark the monitor unlocked by setting the object field to null.
+  //   monitor->set_obj(null);
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/interp_masm_s390.cpp:1111]: nullptr

File: src/hotspot/cpu/s390/interp_masm_s390.cpp [hunk 9]
```diff
@@ -1120,7 +1120,7 @@ void InterpreterMacroAssembler::unlock_object(Register monitor, Register object)
 
   // } else if (Atomic::cmpxchg(obj->mark_addr(), monitor, displaced_header) == monitor) {
   //   // We swapped the unlocked mark in displaced_header into the object's mark word.
-  //   monitor->set_obj(NULL);
+  //   monitor->set_obj(null);
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/interp_masm_s390.cpp:1123]: nullptr

File: src/hotspot/cpu/s390/interp_masm_s390.cpp [hunk 10]
```diff
@@ -1464,13 +1464,13 @@ void InterpreterMacroAssembler::record_klass_in_profile_helper(
...
-//   if (row[0].rec != NULL) {
+//   if (row[0].rec != null) {
 //     // inner copy of decision tree, rooted at row[1]
 //     if (row[1].rec == rec) { row[1].incr(); goto done; }
-//     if (row[1].rec != NULL) {
+//     if (row[1].rec != null) {
 //       // degenerate decision tree, rooted at row[2]
 //       if (row[2].rec == rec) { row[2].incr(); goto done; }
-//       if (row[2].rec != NULL) { count.incr(); goto done; } // overflow
+//       if (row[2].rec != null) { count.incr(); goto done; } // overflow
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/interp_masm_s390.cpp:1473]: nullptr

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 11]
```diff
@@ -3185,10 +3185,10 @@ void MacroAssembler::compiler_fast_lock_object(Register oop, Register box, Regis
   Register zero = temp;
   Register monitor_tagged = displacedHeader; // Tagged with markWord::monitor_value.
   bind(object_has_monitor);
-  // The object's monitor m is unlocked iff m->owner == NULL,
+  // The object's monitor m is unlocked iff m->owner == null,
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:3188]: nullptr

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 12]
```diff
@@ -3388,13 +3388,13 @@ void MacroAssembler::null_check(Register reg, Register tmp, int64_t offset) {
     bind(ok);
   } else {
     if (needs_explicit_null_check((intptr_t)offset)) {
-      // Provoke OS NULL exception if reg = NULL by
+      // Provoke OS null exception if reg = null by
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:3391]: reg is

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 13]
```diff
@@ -3388,13 +3388,13 @@ void MacroAssembler::null_check(Register reg, Register tmp, int64_t offset) {
...
       // accessing M[reg] w/o changing any registers.
       z_lg(tmp, 0, reg);
     }
     // else
       // Nothing to do, (later) access of M[reg + offset]
-      // will provoke OS NULL exception if reg = NULL.
+      // will provoke OS null exception if reg = null.
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:3397]: reg is

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 14]
```diff
@@ -3500,15 +3500,15 @@ void MacroAssembler::encode_klass_not_null(Register dst, Register src) {
 
 // This function calculates the size of the code generated by
 //   decode_klass_not_null(register dst, Register src)
-// when (Universe::heap() != NULL). Hence, if the instructions
+// when (Universe::heap() != null). Hence, if the instructions
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:3503]: nullptr

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 15]
```diff
@@ -3665,8 +3665,8 @@ void MacroAssembler::store_klass_gap(Register s, Register d) {
 // Rop1            - klass in register, always uncompressed.
 // disp            - Offset of klass in memory, compressed/uncompressed, depending on runtime flag.
 // Rbase           - Base address of cKlass in memory.
-// maybeNULL       - True if Rop1 possibly is a NULL.
-void MacroAssembler::compare_klass_ptr(Register Rop1, int64_t disp, Register Rbase, bool maybeNULL) {
+// maybenull       - True if Rop1 possibly is a null.
+void MacroAssembler::compare_klass_ptr(Register Rop1, int64_t disp, Register Rbase, bool maybenullptr) {
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:3669]: `maybe_null` is the best name here, imho.

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 16]
```diff
@@ -3798,9 +3798,9 @@ int MacroAssembler::get_oop_base_complement(Register Rbase, uint64_t oop_base) {
 // Rop1            - Oop in register.
 // disp            - Offset of cOop in memory.
 // Rbase           - Base address of cOop in memory.
-// maybeNULL       - True if Rop1 possibly is a NULL.
-// maybeNULLtarget - Branch target for Rop1 == NULL, if flow control shall NOT continue with compare instruction.
-void MacroAssembler::compare_heap_oop(Register Rop1, Address mem, bool maybeNULL) {
+// maybenull       - True if Rop1 possibly is a null.
+// maybenulltarget - Branch target for Rop1 == null, if flow control shall NOT continue with compare instruction.
+void MacroAssembler::compare_heap_oop(Register Rop1, Address mem, bool maybenullptr) {
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:3803]: make sure these are consistent also

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 17]
```diff
@@ -3914,28 +3914,28 @@ void MacroAssembler::store_heap_oop(Register Roop, const Address &a,
 //
 // only32bitValid is set, if later code only uses the lower 32 bits. In this
 // case we must not fix the upper 32 bits.
-void MacroAssembler::oop_encoder(Register Rdst, Register Rsrc, bool maybeNULL,
+void MacroAssembler::oop_encoder(Register Rdst, Register Rsrc, bool maybenullptr,
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:3917]: Consistency

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 18]
```diff
@@ -3949,7 +3949,7 @@ void MacroAssembler::oop_encoder(Register Rdst, Register Rsrc, bool maybeNULL,
   BLOCK_COMMENT("cOop encoder general {");
   assert_different_registers(Rdst, Z_R1);
   assert_different_registers(Rsrc, Rbase);
-  if (maybeNULL) {
+  if (maybenullptr) {
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:3952]: Consistency

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 19]
```diff
@@ -4050,20 +4050,20 @@ void MacroAssembler::oop_encoder(Register Rdst, Register Rsrc, bool maybeNULL,
 //  - avoid Z_R0 for any of the argument registers.
 //  - keep Rdst and Rsrc distinct from Rbase. Rdst == Rsrc is ok for performance.
 //  - avoid Z_R1 for Rdst if Rdst == Rbase.
-void MacroAssembler::oop_decoder(Register Rdst, Register Rsrc, bool maybeNULL, Register Rbase, int pow2_offset) {
+void MacroAssembler::oop_decoder(Register Rdst, Register Rsrc, bool maybenullptr, Register Rbase, int pow2_offset) {
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:4053]: Consistency

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 20]
```diff
@@ -4074,7 +4074,7 @@ void MacroAssembler::oop_decoder(Register Rdst, Register Rsrc, bool maybeNULL, R
       Label done;
 
       // Rsrc contains a narrow oop. Thus we are sure the leftmost <oop_shift> bits will never be set.
-      if (maybeNULL) {  // NULL ptr must be preserved!
+      if (maybenullptr) {  // null ptr must be preserved!
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:4077]: Consistency

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 21]
```diff
@@ -4134,9 +4134,9 @@ void MacroAssembler::oop_decoder(Register Rdst, Register Rsrc, bool maybeNULL, R
       }
       if (base_preloaded) lgr_if_needed(Rbase_tmp, Rbase);
 
-      // Scale oop and check for NULL.
+      // Scale oop and check for null.
       // Rsrc contains a narrow oop. Thus we are sure the leftmost <oop_shift> bits will never be set.
-      if (maybeNULL) {  // NULL ptr must be preserved!
+      if (maybenullptr) {  // null ptr must be preserved!
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:4139]: Consistency

File: src/hotspot/cpu/s390/macroAssembler_s390.cpp [hunk 22]
```diff
@@ -5526,7 +5526,7 @@ void MacroAssembler::stop(int type, const char* msg, int id) {
 //       should be given for "hand-written" code, if all chain calls are in the same code blob.
 //       Generated code must not undergo any transformation, e.g. ShortenBranches, to be safe.
 address MacroAssembler::stop_chain(address reentry, int type, const char* msg, int id, bool allow_relocation) {
-  BLOCK_COMMENT(err_msg("stop_chain(%s,%s): %s {", reentry==NULL?"init":"cont", allow_relocation?"reloc ":"static", msg));
+  BLOCK_COMMENT(err_msg("stop_chain(%s,%s): %s {", reentry==null?"init":"cont", allow_relocation?"reloc ":"static", msg));
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.cpp:5529]: Glitched patch

File: src/hotspot/cpu/s390/macroAssembler_s390.hpp [hunk 23]
```diff
@@ -771,7 +771,7 @@ class MacroAssembler: public Assembler {
 
   // This function calculates the size of the code generated by
   //   decode_klass_not_null(register dst)
-  // when (Universe::heap() != NULL). Hence, if the instructions
+  // when (Universe::heap() != null). Hence, if the instructions
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.hpp:774]: nullptr

File: src/hotspot/cpu/s390/macroAssembler_s390.hpp [hunk 24]
```diff
@@ -781,8 +781,8 @@ class MacroAssembler: public Assembler {
   static int get_oop_base_pow2_offset(uint64_t oop_base);
   int  get_oop_base(Register Rbase, uint64_t oop_base);
   int  get_oop_base_complement(Register Rbase, uint64_t oop_base);
-  void compare_heap_oop(Register Rop1, Address mem, bool maybeNULL);
-  void compare_klass_ptr(Register Rop1, int64_t disp, Register Rbase, bool maybeNULL);
+  void compare_heap_oop(Register Rop1, Address mem, bool maybenullptr);
+  void compare_klass_ptr(Register Rop1, int64_t disp, Register Rbase, bool maybenullptr);
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.hpp:785]: Consistency

File: src/hotspot/cpu/s390/macroAssembler_s390.hpp [hunk 25]
```diff
@@ -791,20 +791,20 @@ class MacroAssembler: public Assembler {
...
   void store_heap_oop(Register Roop, const Address &a,
                       Register tmp1, Register tmp2, Register tmp3,
                       DecoratorSet decorators = 0);
 
-  void oop_encoder(Register Rdst, Register Rsrc, bool maybeNULL,
+  void oop_encoder(Register Rdst, Register Rsrc, bool maybenullptr,
                    Register Rbase = Z_R1, int pow2_offset = -1, bool only32bitValid = false);
-  void oop_decoder(Register Rdst, Register Rsrc, bool maybeNULL,
+  void oop_decoder(Register Rdst, Register Rsrc, bool maybenullptr,
                    Register Rbase = Z_R1, int pow2_offset = -1);
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/macroAssembler_s390.hpp:808]: Consistency

File: src/hotspot/cpu/s390/nativeInst_s390.cpp [hunk 26]
```diff
@@ -385,7 +385,7 @@ intptr_t NativeMovConstReg::data() const {
 #else
     ShouldNotReachHere();
 #endif
-    return *(intptr_t *)NULL;
+    return *(intptr_t *)nullptr;
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/nativeInst_s390.cpp:388]: Unnecessary cast? Also, this is de-referencing a null pointer, so we might have to do this differently after the conversion.

File: src/hotspot/cpu/s390/runtime_s390.cpp [hunk 27]
```diff
@@ -115,12 +115,12 @@ void OptoRuntime::generate_exception_blob() {
   // Pop the exception blob's C frame that has been pushed before.
   __ z_lgr(Z_SP, saved_sp);
 
-  // [Z_RET]!=NULL was possible in hotspot5 but not in sapjvm6.
+  // [Z_RET]!=null was possible in hotspot5 but not in sapjvm6.
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/runtime_s390.cpp:118]: nullptr

File: src/hotspot/cpu/s390/sharedRuntime_s390.cpp [hunk 28]
```diff
@@ -1043,7 +1043,7 @@ static void object_move(MacroAssembler *masm,
     __ z_stg(rOop, oop_slot_offset, Z_SP);
     __ add2reg(rHandle, oop_slot_offset, Z_SP);
 
-    // If Oop == NULL, use a NULL handle.
+    // If Oop == null, use a null handle.
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/sharedRuntime_s390.cpp:1046]: is

File: src/hotspot/cpu/s390/sharedRuntime_s390.cpp [hunk 29]
```diff
@@ -1324,7 +1324,7 @@ nmethod *SharedRuntime::generate_native_wrapper(MacroAssembler *masm,
                                        stack_slots / VMRegImpl::slots_per_word,
                                        in_ByteSize(-1),
                                        in_ByteSize(-1),
-                                       (OopMapSet *) NULL);
+                                       (OopMapSet *) nullptr);
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/sharedRuntime_s390.cpp:1327]: Unnecessary cast?

File: src/hotspot/cpu/s390/sharedRuntime_s390.cpp [hunk 30]
```diff
@@ -2076,7 +2076,7 @@ static address gen_c2i_adapter(MacroAssembler  *masm,
   // Call patching needed?
   __ load_and_test_long(Z_R0_scratch, method_(code));
   __ z_lg(ientry, method_(interpreter_entry));  // Preload interpreter entry (also if patching).
-  __ z_brne(patch_callsite);                    // Patch required if code != NULL (compiled target exists).
+  __ z_brne(patch_callsite);                    // Patch required if code != null (compiled target exists).
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/sharedRuntime_s390.cpp:2079]: nullptr

File: src/hotspot/cpu/s390/templateInterpreterGenerator_s390.cpp [hunk 31]
```diff
@@ -1125,12 +1125,12 @@ void TemplateInterpreterGenerator::generate_fixed_frame(bool native_call) {
...
   __ z_stg(Z_locals, _z_ijava_state_neg(locals), fp);
 
-  // z_ijava_state->oop_temp = NULL;
+  // z_ijava_state->oop_temp = null;
   __ store_const(Address(fp, oop_tmp_offset), 0);
 
   // Initialize z_ijava_state->mdx.
   Register Rmdp = Z_bcp;
-  // native_call: assert that mdo == NULL
+  // native_call: assert that mdo == null
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/templateInterpreterGenerator_s390.cpp:1133]: nullptr

File: src/hotspot/cpu/s390/templateTable_s390.cpp [hunk 32]
```diff
@@ -4011,8 +4011,8 @@ void TemplateTable::instanceof() {
   }
 
   __ bind(done);
-  // tos = 0: obj == NULL or  obj is not an instanceof the specified klass
-  // tos = 1: obj != NULL and obj is     an instanceof the specified klass
+  // tos = 0: obj == null or  obj is not an instanceof the specified klass
+  // tos = 1: obj != null and obj is     an instanceof the specified klass
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/templateTable_s390.cpp:4015]: nullptr

File: src/hotspot/cpu/s390/templateTable_s390.cpp [hunk 33]
```diff
@@ -4140,7 +4140,7 @@ void TemplateTable::monitorenter() {
     __ bind(exit);
   }
 
-  // Rfree_slot != NULL -> found one
+  // Rfree_slot != null -> found one
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/templateTable_s390.cpp:4143]: nullptr

File: src/hotspot/cpu/s390/vtableStubs_s390.cpp [hunk 34]
```diff
@@ -82,7 +82,7 @@ VtableStub* VtableStubs::create_vtable_stub(int vtable_index) {
   assert(VtableStub::receiver_location() == Z_R2->as_VMReg(), "receiver expected in Z_ARG1");
 
   const Register rcvr_klass   = Z_R1_scratch;
-  address        npe_addr     = __ pc(); // npe == NULL ptr exception
+  address        npe_addr     = __ pc(); // npe == null ptr exception
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/vtableStubs_s390.cpp:85]: This reads a bit strange.

File: src/hotspot/cpu/s390/vtableStubs_s390.cpp [hunk 35]
```diff
@@ -195,7 +195,7 @@ VtableStub* VtableStubs::create_itable_stub(int itable_index) {
 
   // Get receiver klass.
   // Must do an explicit check if offset too large or implicit checks are disabled.
-  address npe_addr = __ pc(); // npe == NULL ptr exception
+  address npe_addr = __ pc(); // npe == null ptr exception
```
Comment (jdksjolen) [inline src/hotspot/cpu/s390/vtableStubs_s390.cpp:198]: This also reads a bit strange

**Review by jdksjolen — COMMENTED** [review]
Reviewed and found a decent chunk of things to fix.

File: src/hotspot/cpu/s390/vtableStubs_s390.cpp (hunk 35 above)
Comment (RealLucy) [reply src/hotspot/cpu/s390/vtableStubs_s390.cpp:198]: Read this as "npe means null pointer exception" or however you would like to spell it.

**Comment by jdksjolen** [comment]
Don't close yet.

File: src/hotspot/cpu/s390/vtableStubs_s390.cpp (hunk 34 above)
Comment (jdksjolen) [reply src/hotspot/cpu/s390/vtableStubs_s390.cpp:85]: So it's saying "NPE is short for null pointer exception"?

**Review by offamitkumar — APPROVED** [review]
I have tested release & fast debug build. Tier1 test on fastdebug as well. PR seems clean. 

Thank you so much for the changes.

**Review by coleenp — APPROVED** [review]
Looks good!

**Comment by jdksjolen** [comment]
Thanks!

/integrate

**Pull request closed without merging**